pkg/health/health.go

```diff
@@ -68,8 +76,17 @@ func (h ConsulHealthChecker) WatchNodeService(nodename string, serviceID string,
 					catalogResults = append(catalogResults, outResult)
 				}
 				kvCheckResult := consulWatchToResult(kvCheck)
-				catalogCheckResult := findWorstResult(catalogResults)
-				resultCh <- findBestResult([]Result{kvCheckResult, catalogCheckResult})
+				catalogCheckResult, HEErr := findWorstResult(catalogResults)
+				if HEErr != nil {
+					resultCh <- kvCheckResult
```

Comment:
GetHealth (on line 64) will return an error if there is no kv result in which case it won't reach this point. tldr kvCheckResult cant be empty here.